repository_service_tuf_worker/repository.py

```diff
@@ -628,6 +633,34 @@ def bootstrap(
 
         return asdict(result)
 
+    def update_settings(
+        self,
+        payload: Dict[str, Any],
+        update_state: Optional[
+            Task.update_state
+        ] = None,  # It is required (see: app.py)
+    ):
+        """Update repository settings with the new settings."""
+        tuf_settings: Dict[str, Any] = payload.get("settings")
+        if tuf_settings is None:
+            raise KeyError("No 'settings' in the payload")
+
+        if tuf_settings.get("expiration") is None:
+            raise KeyError("No 'expiration' in the payload")
+
+        if len(tuf_settings["expiration"]) < 1:
+            raise KeyError("No role provided for expiration policy change")
+
+        logging.info("Updating settings")
+        online_roles = Roles.online_roles()
+        for role in tuf_settings["expiration"]:
+            if role not in online_roles:
+                raise KeyError(f"Role {role} is not a valid dictionary key")
+
+            self.write_repository_settings(
+                f"{role.upper()}_EXPIRATION", tuf_settings["expiration"][role]
+            )
+
```

Comment:
We don't return the task result here in the end.
To the user checking the task status will be `None`